specs/opcode/02MUL_04DIV_06MOD_1bSHL_1cSHR.md

```diff
@@ -60,42 +62,47 @@ $$
 overflow = carry_{hi} + A_1B_3 + A_2B_2 + A_3B_1 + A_2B_3 + A_3B_2 + A_3B_3
 $$
 
-Now back to the opcode circuit for `MUL`, `DIV`, and `MOD`, we first construct
-the `MulAddWordsGadget` with four EVM words `a, b, c, d`.
+Now back to the opcode circuit for `MUL`, `DIV`, `MOD`, `SHL` and `SHR`, we first construct the `MulAddWordsGadget` with four EVM words `quotient, divisor, remainder, dividend`.
 Based on different opcode cases, we constrain the stack pops and pushes as follows
 
-- for `MUL`, two stack pops are `a` and `b`, and the stack push is `d`
-- for `DIV`, two stack pops are `d` and `b`, and the stack push is `a` if `b != 0`; otherwise 0.
-- for `MOD`, two stack pops are `d` and `b`, and the stack push is `c` if `b != 0`; otherwise 0.
+- for `MUL`, two stack pops are `quotient` and `divisor`, and the stack push is `dividend`.
+- for `DIV`, two stack pops are `dividend` and `divisor`, and the stack push is `quotient` if `divisor != 0`; otherwise 0.
+- for `MOD`, two stack pops are `dividend` and `divisor`, and the stack push is `remainder` if `divisor != 0`; otherwise 0.
+- for `SHL`, two stack pops are `quotient` and `shift` when `divisor = 2^shift`, and the stack push is `dividend` if `shift < 256`; otherwise 0.
+- for `SHR`, two stack pops are `dividend` and `shift` when `divisor = 2^shift`, and the stack push is `quotient` if `shift < 256`; otherwise 0.
```

Comment:
I think this will be more accurate, since we need to specify what divisor if when shift >= 256:
"- for `SHR`, two stack pops are `dividend` and `shift` where `divisor = 2^shift` when shift <256 and 0 otherwise. The stack push is `quotient` if `shift < 256` and 0 otherwise."

We need to do the same in the above line for SHL.